tree_test.go

```diff
@@ -1373,3 +1373,34 @@ func BenchmarkTreeLoadAndDelete(b *testing.B) {
 		}
 	})
 }
+func TestLoadVersionForOverwritingCase2(t *testing.T) {
```

Comment:
We should check that all keys have the expected values for the various versions.